core/src/trezor/ui/layouts/tr/__init__.py

```diff
@@ -614,9 +614,11 @@ async def _confirm_ask_pagination(
         ):
             return
 
-        await interact(confirm_more_layout, br_name, br_code, raise_on_cancel=None)
-
-    assert False
+        try:
+            await raise_if_not_confirmed(confirm_more_layout, br_name, br_code)
+            break
+        except ActionCancelled:
+            continue
```

Comment:
if you don't want the ActionCancelled to propagate, don't `raise_if_not_confirmed` -- instead use `interact` and choose what to do based on the return value

Reply:
[7c28cbff83d577f841c951816fc893f11277df54](https://github.com/trezor/trezor-firmware/pull/4346/commits/7c28cbff83d577f841c951816fc893f11277df54)